src/main/java/pinball/PinballComponent.java

```diff
@@ -1,16 +1,249 @@
 package pinball;
 
+import com.badlogic.gdx.math.MathUtils;
 import com.badlogic.gdx.math.Vector2;
-import com.badlogic.gdx.physics.box2d.World;
+import com.badlogic.gdx.physics.box2d.*;
+import com.google.gson.Gson;
+import pinball.DTO.BodiesDTO;
+import pinball.DTO.BodyDTO;
 
 import javax.swing.JComponent;
+import java.awt.*;
+import java.io.*;
+import java.util.ArrayList;
+
 
 public class PinballComponent extends JComponent {
 
+    private long time;
+
+    private static final float WIDTH = 1000;
+    private static final float HEIGHT = 1500;
+
+    private static final float CORNER_LENGTH = 175;
+    private static final float BASE_LENGTH = 350;
+    private static final float FLIPPER_LENGTH = 120;
+
+    private static final float BOX_TO_SCREEN = 10f;
+    private static final float SCREEN_TO_BOX = 1f / BOX_TO_SCREEN;
+
+    private final int radius = 30;
+
     private final World world;
+    private final Body bottom, right, left, top, divider;
+    private final Body topRightCorner, topLeftCorner, bottomRightCorner, bottomLeftCorner;
+    private final Body rightFlipper, leftFlipper;
+    private final Body ball;
+    private ArrayList<Body> bodies = new ArrayList<>();
+
 
-    public PinballComponent() {
+    PinballComponent() {
         world = new World(new Vector2(0, 9.8f), false);
+
+        Gson gson = new Gson();
+
+        try (Reader reader = new FileReader("bodies.json"))
+        {
+            BodiesDTO bodiesDTO = gson.fromJson(reader, BodiesDTO.class);
+            for(BodyDTO bodyDTO : bodiesDTO.bodies)
+            {
+                bodies.add(createBody(bodyDTO));
+            }
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+
+        //set up all horizontal and vertical boxes
+        bottom = createWall(new float[] {100f, (HEIGHT + 100) },
```

Comment:
Why are you creating walls here as well as reading in the bodies from the json?

Reply:
Ah good catch. Seems I hadn't pushed the latest